internal/statesync/syncer.go

```diff
@@ -141,7 +141,14 @@ func (s *syncer) AddSnapshot(peerID types.NodeID, snapshot *snapshot) (bool, err
 
 // AddPeer adds a peer to the pool. For now we just keep it simple and send a
 // single request to discover snapshots, later we may want to do retries and stuff.
-func (s *syncer) AddPeer(peerID types.NodeID) {
+func (s *syncer) AddPeer(peerID types.NodeID) error {
+	var err error
```

Comment:
I believe for this to work you need to put `err` in the return signature:
```suggestion
func (s *syncer) AddPeer(peerID types.NodeID) (err error) {
```

The deferred function can still see this version of `err`, but the value chosen to be returned is already chosen by the time it runs, so modifying this binding will have no effect.